source/extensions/filters/http/grpc_field_extraction/filter_config.h

```diff
@@ -46,6 +48,7 @@ class FilterConfig : public Envoy::Logger::Loggable<Envoy::Logger::Id::filter> {
   absl::flat_hash_map<std::string, std::unique_ptr<Extractor>> proto_path_to_extractor_;
   Protobuf::DescriptorPool descriptor_pool_;
   std::unique_ptr<google::grpc::transcoding::TypeHelper> type_helper_;
+  std::unique_ptr<TypeFinder> type_finder_;
```

Comment:
Seems weird to hold a unique_ptr to a function. Just hold the function itself? it can be empty definition IIUC

Reply:
A const std::function cannot be reassigned to a different callable object. The const qualifier indicates that the object is read-only and its value cannot be modified after initialization. In order to init it later, I have to use unique_ptr.